snap/hooks/configure

```diff
@@ -0,0 +1,28 @@
+#!/bin/sh
+
+
+toml_kvp() {
+	printf "%s = \"%s\"\n" "$1" "$2"
+}
+
+toml_new_section() {
+	printf "\n\n"
+	printf "[%s]\n" "$1"
+}
+
+{
+	echo "# Dynamically generated configuration, do not edit!"
+
+	toml_kvp "hostname" "$(cat /etc/hostname)"
+
+	toml_new_section "aziot_keys"
+	toml_kvp "homedir_path" "$SNAP_COMMON/libaziot_keys_homedir"
+
+	toml_new_section "provisioning"
+	toml_kvp "source" "$(snapctl get provisioning.source)"
+	# TODO: don't print this if not set
+	toml_kvp "connection_string" "$(snapctl get provisioning.connection-string)"
+
+} > $SNAP_DATA/etc/aziot/config.toml
+
+$SNAP/bin/aziotctl config apply
```

Comment:
Is there any way that someone could pass as a config value a file path (or the full content) of an existing config.toml and then this configure hook could just call `$SNAP/bin/aziotctl config apply -c 'path to file'`?

Reply:
We could certainly implement something along those lines. 

`snap set azure-iot-identity config-file-path='path to file'` -> `$SNAP/bin/aziotctl config apply -c 'path to file'` would be a bit limited. We could do it as long as `'path to file'` is in the users home directory, but currently there isn't a very good way to give access to arbitrary file paths (and plus for unattended install/config you'd have to somehow seed that file onto the device, and there aren't really users and home directories on ubuntu core, and so this path gets a bit messy for cases besides development work).

More interesting is the idea of doing something more like `snap set azure-iot-identity raw-config-file="$(cat 'path to config file')" -> `snapctl get raw-config-file > /tmp/cfg.toml ; $SNAP/bin/aziotctl config apply -c /tmp/cfg.toml`. Then, on unattended installs, you could also just pass in the raw file contents as a multiline default config value.

As a note, this config hook is probably the least developed part of this whole effort so far, and definitely needs quite a bit more work.

Reply:
I think we should start with the raw-config-file option to make setting the config as much of a pass-through operation as possible.  Otherwise, I can see a lot of effort going into ensuring that the range of allowed options in the config.toml can also be expressed via `snap set azure-iot-identity ...` not to mention that whenever we decided to add/remove config options then this configure script would also need to be modified.